lib/rubocop/cop/rails/eager_evaluation_log_message.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks that blocks are used for interpolated strings passed to
+      # `Rails.logger.debug`.
+      #
+      # By default, Rails production environments use the `:info` log level.
+      # At the `:info` log level, `Rails.logger.debug` statements do not result
+      # in log output. However, Ruby must eagerly evaluate interpolated string
+      # arguments passed as method arguments. Passing a block to
+      # `Rails.logger.debug` prevents costly evaluation of interpolated strings
+      # when no output would be produced anyway.
+      #
+      # @example
+      #   #bad
+      #   Rails.logger.debug "The time is #{Time.zone.now}."
+      #
+      #   #good
+      #   Rails.logger.debug { "The time is #{Time.zone.now}." }
+      #
+      class EagerEvaluationLogMessage < Base
+        extend AutoCorrector
+
+        MSG = 'Pass a block to `Rails.logger.debug`.'
```

Comment:
Can you add `RESTRICT_ON_SEND` for improved performance?

```suggestion
        MSG = 'Pass a block to `Rails.logger.debug`.'
        RESTRICT_ON_SEND = %i[debug].freeze
```